### PR TITLE
Code Review Dabesu 2

### PR DESCRIPTION
**Pull request by Dabesu**
### Analysis of the program.
Overall the program is nice for the first generation pokemon, but in question is just a simplified serebii, Maybe consider adding something that serebii doesn't offer or update something that they already have.
### Was the program available in UC Github on time?
Yes.
### Is the program documented/commented well enough for you to understand?
Yes
### Does the program compile?
Yes

- In Bootstrap, navbar toggler hides the nav when the view port is smaller then a normal webpage.

- In jQuery, things like document.getElementsByClassName can be shortened to $('.pokemon');

- In jQuery, the hashtag in $('#filter') is a shortcut for ID in this case document.getElementsById

### CODE REVIEW

File: src/main/resources/templates/index.html [hunk 1]
```diff
@@ -13,10 +13,10 @@
 
 <nav class="navbar navbar-expand-lg navbar-light bg-light">
     <a class="navbar-brand">PokemonViewer</a>
-    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarTogglerDemo02" aria-controls="navbarTogglerDemo02" aria-expanded="false" aria-label="Toggle navigation">
+    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarToggler" aria-controls="navbarToggler" aria-expanded="false" aria-label="Toggle navigation">
```
Comment (Dabesu) [inline src/main/resources/templates/index.html:13]: Updated navbarTogglerDemo02 to navbarToggler to make the code cleaner, must have been left on from a tutorial.

File: src/main/resources/templates/index.html [hunk 2]
```diff
@@ -53,17 +53,17 @@
...
-        let input = document.getElementById('filter').value;
+        let input = $('#filter').val();
         input = input.toLowerCase();
-        let x = document.getElementsByClassName('pokemon');
-
-        for (i = 0; i < x.length; i++) {
-            if (!x[i].innerHTML.toLocaleLowerCase().includes(input)) {
-                x[i].style.display = "none";
+        let allPokemon = $('.pokemon');
+        allPokemon.each((index,data)=>{
```
Comment (Dabesu) [inline src/main/resources/templates/index.html:57]: changed "x" to allPokemon as the variable x can be confusing.

File: src/main/resources/templates/index.html [hunk 3]
```diff
@@ -53,17 +53,17 @@
...
+            data = $(data)
+            if (!data.html().toLowerCase().includes(input)){
+                data.toggle(false)
```
Comment (Dabesu) [inline src/main/resources/templates/index.html:60]: Converted the for loop to use jQuery functions.

**Comment by discospiff** [comment]
Several good suggestions, like renaming x to pokemon.  It's worth reviewing and considering what to merge.

**Pull request merged**